src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java

```diff
@@ -2,34 +2,48 @@
-        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getTo()).ifPresent(island -> {
-                    if (IridiumSkyblock.getInstance().getIslandManager().isBannedOnIsland(island, user)) {
-                        event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().youHaveBeenBanned
-                                .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
-                                .replace("%owner%", island.getOwner().getName())
-                                .replace("%name%", island.getName())
-                        ));
-                        event.setCancelled(true);
-                    } else {
-                        Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () ->
-                                        PlayerUtils.sendBorder(event.getPlayer(), island)
-                                , 1);
-                    }
+        Optional<Island> islandOptional = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getTo());
+        if (islandOptional.isPresent()) {
+            Island island = islandOptional.get();
+            if (!event.getPlayer().hasPermission("iridiumskyblock.bypassban") && IridiumSkyblock.getInstance().getIslandManager().isBannedOnIsland(island, user)) {
+                event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().youHaveBeenBanned
+                        .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)
+                        .replace("%owner%", island.getOwner().getName())
+                        .replace("%name%", island.getName())
+                ));
+                event.setCancelled(true);
+            } else {
+                if (island.isVisitable() || (user.isBypassing() || island.getMembers().contains(user) || island.getOwner().equals(user))) {
+                    Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () ->
+                                    PlayerUtils.sendBorder(event.getPlayer(), island)
+                            , 1);
+                } else {
+                    event.setCancelled(true);
+                    event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().islandIsPrivate
+                            .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
                 }
-        );
+            }
+        } else {
+            event.getPlayer().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().noIslandFound
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+            if (!event.getPlayer().hasPermission("iridiumskyblock.locationisland.bypass")) event.setCancelled(true); // Unsafe Teleportation
```

Comment:
What exactly is the point of this code? What was the exploit you were talking about?

Reply:
For reproduice : go on an island open
use /sethome
Now close this island.
Now all player has /sethome your island can bypass close island